src/docs/patterns/loading.mdx

```diff
@@ -0,0 +1,123 @@
+
+The placeholder should match as closely as possible the real structure of the data. For example, if you are loading a list where each item has an icon, two lines of text, and an action button, then your placeholder should have a circle, two rectangles for the text, and another rectangle for the button. You should show as many placeholder elements as necessary to fill the area in the application that is loading data.
+
+There are a number of libraries available to help you create this placeholder content, such as:
+
+-   [@netbasal/ngx-content-loader](https://www.npmjs.com/package/@netbasal/ngx-content-loader) (Angular, Ionic)
+-   [ion-skeleton-text](https://ionicframework.com/docs/api/skeleton-text) (Ionic)
+-   [placeholder-loading](https://www.npmjs.com/package/placeholder-loading) (React, Angular, Ionic)
+-   [react-content-loader](https://www.npmjs.com/package/react-content-loader) (React)
+-   [rn-placeholder](https://www.npmjs.com/package/rn-placeholder) (React Native)
+
+<React.Fragment>
+    <ImageGrid
+        images={[Image1, Image2]}
+        caption={'When data is loading, use placeholder shapes for where the data will appear.'}
+    />
+    <DemoCard
+        repository={'loading-states'}
+        angular={
+            'https://stackblitz.com/github/pxblue/angular-design-patterns?file=src%2Fapp%2Fpages%2Floading-states%2Floading-states.component.ts'
+        }
+        react={
+            'https://codesandbox.io/s/github/pxblue/react-design-patterns/tree/master?file=/src/pages/loading-states/index.tsx&initialpath=loading-states'
+        }
+        ionic
+        reactNative
+    />
+</React.Fragment>
+
+### Full Screen
+
+Full screen progress indicators can be used when transitioning between multiple screens in a workflow (such as user registration) or refreshing / reloading a particular element on a page. These make use of a semi-transparent overlay with a large, centered spinner. This variation of indicator is useful when you want to block user interaction with the screen (or a section of the screen) while processing occurs.
+
+TODO: Image?
+
+### Inline Spinners
+
+Inline spinners can be used as an alternative to full-screen loading indicators. The most common application of inline spinners is during [Form](/patterns/forms) submission / verification. If you need to wait for a server response to validate inputs in a form (after clicking the submit button), you should disable the submit button and replace its label with a spinner. If the verification succeeds, the application should proceed to the next screen. If verification fails, the appropriate error messages should be displayed and the submit button label restored.
```

Comment:
I think the submit, next, etc. button always needs to be disabled when going to the next screen to prevent double-taps. I'd be fine with not using disabled styling though (e.g., still use a blue[500] button, but clicking it while it's showing the spinner won't send the request again).